geonode/base/api/tests.py

```diff
@@ -2658,6 +2661,123 @@ def test_only_get_method_is_available(self):
         response = self.client.put(url)
         self.assertEqual(response.status_code, 403)
 
+    def test_insert_one_linked_resource(self):
+        url = reverse("base-resources-linked_resources", args=[self.doc.id])
+
+        self.client.force_login(get_user_model().objects.get(username="admin"))
+
+        response = self.client.post(url, data={"target": self.map.id})
+
+        link_connected = LinkedResource.objects.get(source_id=self.doc.id)
+
+        self.assertEqual(response.status_code, 201)
+
+        self.assertEqual(self.doc.id, link_connected.source_id)
+
+        self.assertEqual(self.map.id, link_connected.target_id)
+
+    def test_insert_self_as_linked_resource(self):
+        self.client.force_login(get_user_model().objects.get(username="admin"))
+        url = reverse("base-resources-linked_resources", args=[self.doc.id])
+
+        # linked resource cannot be linked to itself
+        response = self.client.post(url, data={"target": self.doc.id})
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_insert_existing_linked_resource(self):
+        url = reverse("base-resources-linked_resources", args=[self.doc.id])
+        self.client.force_login(get_user_model().objects.get(username="admin"))
+        LinkedResource.objects.create(source_id=self.doc.id, target_id=self.doc.id)
+
+        # linked resource cannot be duplicated
+        response = self.client.post(url, data={"target": self.doc.id})
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_insert_multiple_linked_resource(self):
+        url = reverse("base-resources-linked_resources", args=[self.doc.id])
+
+        self.client.force_login(get_user_model().objects.get(username="admin"))
+
+        response = self.client.post(url, data={"target": self.map.id})
+        response = self.client.post(url, data={"target": self.dataset.id})
+
+        list_connected = LinkedResource.objects.filter(source_id=self.doc.id).all()
+
+        list_connected_targets = [linked.target_id for linked in list_connected]
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(2, len(list_connected))
+
+        self.assertTrue((self.map.id in list_connected_targets))
+        self.assertTrue((self.dataset.id in list_connected_targets))
+
+    def test_insert_invalid_linked_resource(self):
+
+        url = reverse("base-resources-linked_resources", args=[self.doc.id])
+
+        self.client.force_login(get_user_model().objects.get(username="admin"))
+
+        # generate an invalid id
+        invalid_id = max([r.id for r in ResourceBase.objects.all()]) + 1
```

Comment:
Using `last` has a better performace `ResourceBase.objects.last() + 1` since you dont have to read the whole queryset